### PR TITLE
check for macOS

### PR DESCRIPTION
**Pull request by JayBrown**
haven't tried to install your fork of Diogenes on macOS yet, but sha256sum vs. shasum was an issue in the original by @nickjwhite

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -18,18 +18,37 @@ UNICODEVERSION = 7.0.0
...
+UNAME=$(uname -s)
+case "${UNAME}" in
+	Linux*)     SYSTEM=Linux;;
+	Darwin*)    SYSTEM=Mac;;
+	CYGWIN*)    SYSTEM=Cygwin;;
+	MINGW*)     SYSTEM=MinGW;;
+	*)          SYSTEM="${UNAME}"
+esac
+
+
```
Comment (Doc73) [inline makefile]: I have errors Linux at these lines. Please, test in MacOS:
```
ifeq ($(OS),Windows_NT)
    detected_OS := Windows
else
    detected_OS := $(shell uname -s)
endif

ifeq ($(detected_OS),Linux)
	$SYSTEM=Linux
endif
ifeq ($(detected_OS),Darwin)
	SYSTEM=Mac
endif
```
In this way, I have error in sequent `if ... else` sections

**Comment by JayBrown** [comment]
Yes, that was my original issue. MacOS only has the default BSD/POSIX shasum, but on my system another shasum was installed with perl v5.28.0, which works the same way as the POSIX version. So on macOS `shasum -a 256` definitely works. But I can't speak for other systems, i.e. can't say if `shasum -a 256 <path>` works on Linux, Solaris etc. If it does, you can roll back my pull requests and just change the checksumming command instead.

You might want to change wget to curl too in the scripts for cross-platform (see my other post). I'll then test it later today.

**Comment by Doc73** [comment]
Dear all, and expecially @JayBrown , could you test branch `check_os`, please? 
I try to solve the issue of checking the OS (Linux/Mac). In my Linux System it works fine!

**Comment by nickjwhite** [comment]
Great, <code>shasum -a 256</code> definitely works on (at least) Linux and FreeBSD, so I think it's reliable enough to use everywhere.

**Comment by Doc73** [comment]
@nickjwhite 
I can confirm `shasum -c -a 256` works very well on Linux. Therefore, we can remove the conditional sections, at least for Mac/GNU Linux. Do you know how it works on Windows?

**Comment by nickjwhite** [comment]
The build system is designed to build a Windows package from a unix-alike, like Linux. It probably would still work in Windows, with something like Cygwin, and in that case it should work just the same.

**Pull request merged**